user_tools/src/spark_rapids_tools/utils/util.py

```diff
@@ -248,10 +248,21 @@ def get_system_memory_in_gb(cls) -> int:
         ps_memory = psutil.virtual_memory()
         return int(0.8 * ps_memory.total / (1024 ** 3))
 
+    @classmethod
+    def get_max_jvm_threads(cls) -> int:
+        """
+        Get the total cpu_count.
+        """
+        # Maximum number of threads that can be used in the tools JVM.
+        # cpu_count returns the logical number of cores. So, we take a 50% to get better representation
+        # of physical cores.
+        return min(6, (psutil.cpu_count() + 1) // 2)
```

Comment:
Curious why it is 6 in the first argument.

Reply:
Can we use `psutil.cpu_count(logical=False)` to get number of physical cores?

Reply:
It is a hardcoded value suggesting that even with enough memory resources, we still want to be cautios about the number of threads running concurrently.
Assuming one of the eventlog is really huge, then 1 thread will starve the heap.
That's why we want to limit the number of threads to reduce the probability of concurrent threads processing huge files at the same time.